psd.h

```diff
@@ -12,6 +12,10 @@
 #include <cstdio>
 #include <string>
 #include <vector>
+#include <memory>
+
+#undef TRACE
+#define TRACE std::printf
```

Comment:
Don't define `TRACE()` in a public header file like `psd.h`. And define it just as it was: `TRACE(...)` (you can have your local change to `TRACE std::printf` but don't commit it).